### PR TITLE
Convert panics to errors

### LINKED ISSUE #7: Inconsistently returning errors or panicking
Several functions in `lms/common/types.go` inconsistently return errors or panic. I expect the behavior should be always return an error here, given that these functions (e.g., `LmsParams`) are called on arbitrary bytes.

https://github.com/trailofbits/lms-go/blob/c7cd3adfe891e90a504025011b6a0e04921b7d68/lms/common/types.go#L113-L128
https://github.com/trailofbits/lms-go/blob/c7cd3adfe891e90a504025011b6a0e04921b7d68/lms/common/types.go#L131-L146
https://github.com/trailofbits/lms-go/blob/c7cd3adfe891e90a504025011b6a0e04921b7d68/lms/common/types.go#L149-L229

### PR DESCRIPTION
**Pull request by tob-scott-a**
Fixes #7

### CODE REVIEW

File: lms/common/types.go [hunk 1]
```diff
@@ -119,10 +119,17 @@ func (x lms_type_code) LmsType() (lms_type_code, error) {
...
+		if err != nil {
+			return 0, err
+		}
+		otssiglen, err := otstc.LmsOtsSigLength()
+		if err != nil {
+			return 0, err
+		}
+		return uint64(4 + 4 + otssiglen + (params.H * params.M)), nil
 	} else {
 		panic("LmsSigLength(): invalid type code")
```
Comment (woodruffw) [inline lms/common/types.go]: There's another panic here -- do we want to remove it as well?

File: lms/common/types.go [hunk 2]
```diff
@@ -138,54 +145,58 @@ func (x lms_type_code) LmsOtsType() (lms_type_code, error) {
...
+func (x lms_type_code) LmsOtsSigLength() (uint64, error) {
 	if x >= LMOTS_SHA256_N32_W1 && x <= LMOTS_SHA256_N32_W8 {
-		return x.Params().SIG_LEN
+		params, err := x.Params()
+		if err != nil {
+			return 0, err
+		}
+		return params.SIG_LEN, nil
 	} else {
 		panic("LmsOtsSigLength(): invalid type code")
```
Comment (woodruffw) [inline lms/common/types.go]: Same `panic` question 🙂

**Review by woodruffw — COMMENTED** [review]
Overall approach LGTM, two questions about remaining panics 🙂

File: lms/common/types.go (hunk 1 above)
Comment (tob-scott-a) [reply lms/common/types.go]: Yes.

File: lms/common/types.go (hunk 2 above)
Comment (tob-scott-a) [reply lms/common/types.go]: Yes.

**Review by incertia — APPROVED** [review]

**Pull request merged**